### DISCUSSION TITLE
Changing keyboard layout

### DISCUSSION THREAD

**Discussion by tovare** [community/community#30714] [Codespaces]
Hi,

I need to install a keyboard layout for one of the Scandinavian language's, preferably Norwegian. 

In VS code I see no option for any of these languages, and I have been unsuccessful in finding any tips that works from the command line.

( I run codespaces via a terminal program called Blink on my iPad. The keyboard is a Scandinavian ISO layout Logitech MX Mechanical Mini )

Any advice is appreciated.

**Comment by osortega** [comment]
Hello @tovare! 👋 
Could you help elaborating more on what you are trying to do? Is it localization of VS Code or being able to map keybindings to different keyboard layouts?

If it's the first one, VS Code does not support translation to Norwegian language yet. You can find more about supported languages and how to contribute [here](https://code.visualstudio.com/docs/getstarted/locales).

If it's the latter, within VS Code you can go to `Code` -> `Preferences` -> `Keyboard Shortcuts` and map specific keys of your keyboard layout to a particular action. Might be worth mentioning that VS Code has an opened feature request to support different keybindings based on custom keyboard layout [here](https://github.com/microsoft/vscode/issues/1240).

Looks like you are also having issues when using Codespaces from a terminal program, in that case that program might need to be the one that adds support for different keyboard layouts.

**Reply by tovare** [reply]
Thank you for answering this. I poked around further and found that in the settings for hardware keyboards it is possible to set the layout for both Norwegian and English, by setting the English (USA) layout to Norwegian it worked.

The thing that tripped med up a bit is that the VS Code the bottom status bar showed a  quick-selector for a multitude of layouts for different languages, but not any Scandinavian ones. I think that you are right that the terminal program Blink might also be a place to configure this. But having the layout for English language set to Norwegian layout is fine :-)

Regards
